### PR TITLE
Support order option in astype method

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented order option support in `astype` method.
It uses `elementwise_copy`. Is it better to use `ndarray.copy_from_device`?

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -260,11 +260,19 @@ cdef class ndarray:
...
 
         Args:
             dtype: Type specifier.
+            order ({'C', 'F', 'A', 'K'}): Row-major (C-style) or column-major
+                (Fortran-style) order.
+                When `order` is 'A', it uses 'F' if `a` is column-major and
+                uses `C` otherwise.
+                And when `order` is 'K', it keeps strides as closely as
+                possible.
+                This function currently does not support order 'K'.
```
Comment (okuta) [inline cupy/core/core.pyx]: Is this correct?

Comment (unnonouno) [inline cupy/core/core.pyx]: Oh, sorry.

**Comment by okuta** [comment]
LGTM!

**Comment by okuta** [comment]
> Is it better to use ndarray.copy_from_device?

No. We have to fix caller method.

**Pull request merged**